### ISSUE TITLE
Archive POST query.

### ISSUE THREAD

**Issue by hungrymonkey** [webrecorder/pywb#462]
## Is your feature request related to a problem? Please describe.

https://www.smgov.net/departments/clerk/agendas.aspx
At the city of Santa Monica website, I can query the current year with a simple GET request, but I need to send the current session token with a POST request to query the previous years

## Describe the solution you'd like

I need a way to archive a POST generated query.  

## Describe alternatives you've considered

I attempted warcio but the website isnt replayable. It contains all the html elements but it is missing all other components

<img width="1792" alt="Screen Shot 2019-04-15 at 10 40 14 PM" src="https://user-images.githubusercontent.com/771339/56224658-d16a3b00-6024-11e9-8bee-998874346528.png">

**Comment by N0taN3rd** [comment]
@hungrymonkey could you provide some more context as to how the page was captured? 

I.e. did you use Webrecorder, Heritrix, or somethings else?

**Comment by hungrymonkey** [comment]
```
def parse_session(html, year):
	soupget = BeautifulSoup(html, 'html.parser')
	form = soupget.find('form', {'name': 'aspnetForm'})
	formInputs = form.findChildren('input')
	payload = dict()
	for input in formInputs:
		if input.get('name') in ["EktronClientManager", "__VIEWSTATE", "__VIEWSTATEGENERATOR", "__EVENTVALIDATION"]:
			payload[input.get('name')] = input.get('value')
	payload["ctl00$ctl00$bodyContent$mainContent$ddlYears"] = str(year)

if __name__ == "__main__":
	config.initalize_project_root()
	today = time.strftime("%Y%m%d%H%M%S", time.gmtime())
	#config.initalize_record()
	r = requests.get( config.AGENDA_URL)
	session_headers = parse_session(r.text, 2018)
	fileName = 'rec-' + today + '-psuedos-MacBook-Pro.local.warc.gz'
	with open( fileName , 'wb') as output:
		writer = WARCWriter(output, gzip=True)
		response = requests.post(config.AGENDA_URL, data=session_headers)
		headers_list = response.raw.headers.items()
		http_headers = StatusAndHeaders('200 OK', headers_list, protocol='HTTP/1.0')
		record = writer.create_warc_record(config.AGENDA_URL, 'response', payload=response.raw, http_headers=http_headers)
		writer.write_record(record)
```
Here is a small snippet of the code. I am just using warcio by itself.
https://github.com/webrecorder/warcio/issues/60
By design, it doesn't support replay.

May I ask how to use Pywb to send a post command on page load? This website need a request to obtain a session token then resend it to change the year. I attach an example of the token below

[example_token.txt](https://github.com/webrecorder/pywb/files/3086038/example_token.txt)

**Comment by N0taN3rd** [comment]
> May I ask how to use Pywb to send a post command on page load? This website need a request to obtain a session token then resend it to change the year. I attach an example of the token below

@hungrymonkey you would need to ensure that POST request is archived, e.g. inside the WARC for the page.

Since using warcio as the means to preserve the page it can not preserve that required POST request unless you make it yourself, i.e. from your python code, or you use Webrecorder / pywb's built in recording mode. Both ways would work but the former (using Webrecorder) would give you the [desired outcome 100%](https://webrecorder.io/jberlin/bug-reports/list/pywb-issue-462).

**Comment by ikreymer** [comment]
Yes, pywb is designed to be used through a browser.. It will capture and should be capable of replaying POST requests as well, it will send the same traffic (there is also support for fuzzy matching as well if certain params need to be removed).

You can start a headless browser and have it load a page through pywb.
Are you trying to automate logging into a site, or just do it once?

For using warcio, there's also a simpler syntax now using the `capture_http` method available in the latest release:
https://github.com/webrecorder/warcio#quick-start-to-writing-a-warc

**Comment by hungrymonkey** [comment]
@N0taN3rd  The example you gave me isnt the desired result. I need to preserve 1988-2018 agenda. The only way to access those pages through the post request. The site is a server rendered application.

@ikreymer I am not logging into the site at all. I need to resend a session token embedded within the html provided on first page load. Without the token, I cannot load previous agenda items. Gotta love government contracted php.

**Comment by N0taN3rd** [comment]
@hungrymonkey have you tried in recording mode? That is to say you would navigate your browser, after being configured to use pywb as its proxy, to the pages you desire preserved. 

Can do that now using [webrecorder.io](https://webrecorder.io) if setting things up locally is proving to be to difficult.

**Comment by hungrymonkey** [comment]
@N0taN3rd  pywb seems to capture the first load. Like I said, I need to capture a server generated POST query. If I can load a Post then capture it, can you tell me how to do it? I tried digging into pywb docs.

**Comment by hungrymonkey** [comment]
May I ask what does patch url do in webrecorder.io?
It seems like selecting patch then record seems to capture the state I want.

**Comment by hungrymonkey** [comment]
I attempted to use proxys but any POST generated archive is invisible to pywb

```
proxy:
  coll: smgov
  recording: true
```
[rec-20190507020217371784-psuedos-MacBook-Pro.local.warc.gz](https://github.com/webrecorder/pywb/files/3150489/rec-20190507020217371784-psuedos-MacBook-Pro.local.warc.gz)



```
r = requests.get( config.AGENDA_URL)
	session_headers = parse_session(r.text, 2018)
	try:
		requests.request('POST', config.AGENDA_URL, data=session_headers, proxies={'https' : config.RECORD_URL}, verify=False)
		time.sleep(5)
	except Exception:
		traceback.print_exc()
		print("Error proxy post " , file=sys.stderr)
```

**Comment by hungrymonkey** [comment]
Nevermind, I change the search options and found it.

**Comment by N0taN3rd** [comment]
@hungrymonkey apologies for the delay in my reply. Before coming to this solution did you check to see if the headers you are intercepting were being rewritten and included properly by pywb

**Comment by hungrymonkey** [comment]
Ok, I am taking a look and the warc didnt save the post request. I will look again.

**Comment by N0taN3rd** [comment]
To clarify was the page, when recorded via proxy mode, did it and its JS receive the headers properly

**Comment by hungrymonkey** [comment]
I do not think so. The site recorded the current year instead of 2018. I did not think the site did not see the POST headers
